lib/actions-oclif/preload.ts

```diff
@@ -0,0 +1,549 @@
+		After preloading, the balenaOS image file can be flashed to a device's SD card.
+		When the device boots, it will not need to download the application, as it was
+		preloaded.
+
+		Warning: "balena preload" requires Docker to be correctly installed in
+		your shell environment. For more information (including Windows support)
+		check: https://github.com/balena-io/balena-cli/blob/master/INSTALL.md
+	`;
+
+	public static examples = [
+		'$ balena preload balena.img --app 1234 --commit e1f2592fc6ee949e68756d4f4a48e49bff8d72a0 --splash-image image.png',
+		'$ balena preload balena.img',
+	];
+
+	public static args = [
+		{
+			name: 'image',
+			description: 'the image file path',
+			required: true,
+		},
+	];
+
+	public static usage = 'preload <image>';
+
+	public static flags: flags.Input<FlagsDef> = {
+		app: flags.string({
+			description: 'name, slug or numeric ID of the application to preload',
+			char: 'a',
+		}),
+		commit: flags.string({
+			description: `\
+The commit hash for a specific application release to preload, use "current" to specify the current
+release (ignored if no appId is given). The current release is usually also the latest, but can be
+manually pinned using https://github.com/balena-io-projects/staged-releases .\
+`,
+			char: 'c',
+		}),
+		'splash-image': flags.string({
+			description: 'path to a png image to replace the splash screen',
+			char: 's',
+		}),
+		'dont-check-arch': flags.boolean({
+			description:
+				'disables check for matching architecture in image and application',
+		}),
+		'pin-device-to-release': flags.boolean({
+			description:
+				'pin the preloaded device to the preloaded release on provision',
+			char: 'p',
+		}),
+		'add-certificate': flags.string({
+			description: `\
+Add the given certificate (in PEM format) to /etc/ssl/certs in the preloading container.
+The file name must end with '.crt' and must not be already contained in the preloader's
+/etc/ssl/certs folder.
+Can be repeated to add multiple certificates.\
+`,
+		}),
+		...dockerConnectionCliFlags,
+		// Redefining --dockerPort here (defined already in dockerConnectionCliFlags)
+		// without -p alias, to avoid clash with -p alias of pin-device-to-release
+		dockerPort: flags.integer({
+			description:
+				'Docker daemon TCP port number (hint: 2375 for balena devices)',
+			parse: (p) => parseAsInteger(p, 'dockerPort'),
+		}),
+		// Not supporting -h for help, because of clash with -h in DockerCliFlags
+		// Revisit this in future release.
+		help: flags.help({}),
+	};
+
+	public static authenticated = true;
+
+	public static primary = true;
+
+	public async run() {
+		const { args: params, flags: options } = this.parse<FlagsDef, ArgsDef>(
+			PreloadCmd,
+		);
+
+		const balena = getBalenaSdk();
+		const balenaPreload = await import('balena-preload');
+		const visuals = getVisuals();
+		const nodeCleanup = await import('node-cleanup');
+		const { ExpectedError, instanceOf } = require('../errors');
```

Comment:
```suggestion
		const { ExpectedError, instanceOf } = await import('../errors');
```

Reply:
Ahgr another

Reply:
Actually, I've just realised that we have a bunch of `exitWithExpectedError()`s, which should changed to `throw new ExpectedError()`s.  So probably the `ExpectedError` import should then be moved to the top.

Reply:
Sure, if there's any performance impact to that I'll catch/deal with it later - finishing the conversion and removing capitano will more than offset it anyway :)